src/Hazelcast.Net/DistributedObjects/DistributedObjectFactory.cs

```diff
@@ -57,6 +58,49 @@ public DistributedObjectFactory(Cluster cluster, SerializationService serializat
             _logger = loggerFactory.CreateLogger<DistributedObjectFactory>();
         }
 
+        /// <summary>
+        /// Gets all distributed objects.
+        /// </summary>
+        /// <param name="cancellationToken">A cancellation token.</param>
+        /// <returns>The distributed objects.</returns>
+        public async Task<ICollection<DistributedObjectInfo>> GetAllAsync(CancellationToken cancellationToken = default)
+        {
+            // note: this method implements the logic from Java HazelcastClientInstanceImpl.getDistributedObjects method
+
+            // gather locally-known objects
+            var objects = new HashSet<DistributedObjectInfo>();
+            await foreach (var (info, _) in _objects) objects.Add(info);
+
+            // fetch remotely-known objects
+            var requestMessage = ClientGetDistributedObjectsCodec.EncodeRequest();
+            var responseMessage = await _cluster.Messaging.SendAsync(requestMessage, cancellationToken).CfAwait();
+            var response = ClientGetDistributedObjectsCodec.DecodeResponse(responseMessage);
+
+            var result = new List<DistributedObjectInfo>();
+
+            foreach (var info in response.Response)
+            {
+                objects.Remove(info); // remove remotely-known objects
+
+                // at that point Java would actually make sure that the client has proxies for
+                // each object that was reported by the cluster - however that *cannot* work in
+                // .NET due to strong generics, because we cannot, for instance, create a
+                // HMap<TKey, TValue> object, since we don't know TKey nor TValue. this is not
+                // an issue really, proxies will be created on-demand.
+
+                result.Add(info); // build result
+            }
+
+            // locally destroy objects that are not remotely-known
```

Comment:
I know it the logic is from Java but removing the local object is odd. Invoking a read only method modifies the client :S It should be idempotent, right? However, most probably, this is the reason that this method is not much promoted.

Reply:
agree, but I've tried to stay close to Java (for once)